### PR TITLE
r/aws_connect_quick_connect

### PR DESCRIPTION
**Pull request by GlennChia**
### Community Note

* Please vote on this pull request by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original pull request comment to help the community and maintainers prioritize this request
* Please do not leave "+1" or other comments that do not add relevant new information or questions, they generate extra noise for pull request followers and do not help prioritize the request

Relates #21022

Output from acceptance testing:

```
$ make testacc TESTS=TestAccConnectQuickConnect_serial PKG=connect
\==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./internal/service/connect/... -v -count 1 -parallel 20 -run='TestAccConnectQuickConnect_serial' -timeout 180m
=== RUN   TestAccConnectQuickConnect_serial
=== RUN   TestAccConnectQuickConnect_serial/basic
=== RUN   TestAccConnectQuickConnect_serial/disappears
--- PASS: TestAccConnectQuickConnect_serial (296.54s)
    --- PASS: TestAccConnectQuickConnect_serial/basic (190.81s)
    --- PASS: TestAccConnectQuickConnect_serial/disappears (105.73s)
PASS
ok      github.com/hashicorp/terraform-provider-aws/internal/service/connect    304.229s

...
```

### CODE REVIEW

File: internal/service/connect/quick_connect.go [hunk 1]
```diff
@@ -0,0 +1,420 @@
...
+	return &schema.Resource{
+		CreateContext: resourceQuickConnectCreate,
+		ReadContext:   resourceQuickConnectRead,
+		UpdateContext: resourceQuickConnectUpdate,
+		DeleteContext: resourceQuickConnectDelete,
+		Importer: &schema.ResourceImporter{
+			StateContext: schema.ImportStatePassthroughContext,
+		},
+		Timeouts: &schema.ResourceTimeout{
+			Create: schema.DefaultTimeout(connectQuickConnectCreatedTimeout),
```
Comment (AdamTylerLynch) [inline internal/service/connect/quick_connect.go:29]: You provide Timeouts, and the ability to override via Configuration, but do not have corresponding wait() and status() functions in the resource implementation. After reviewing the Connect service, I see this is actually pretty common, meaning that even though we have Timeout values exposed to users, we don't actually honor them.

The question becomes, do we actually need to provide timeouts? These are typically reserved for resources that have eventual consistency/asynchronous APIs.

See a good example of honoring timeouts here: https://github.com/hashicorp/terraform-provider-aws/blob/f445f5e1980a8b4a6581b2964b8ba70b80330128/internal/service/lexmodels/bot.go#L268

File: internal/service/connect/wait.go [hunk 2]
```diff
@@ -20,6 +20,9 @@ const (
 
 	connectHoursOfOperationCreatedTimeout = 5 * time.Minute
```
Comment (AdamTylerLynch) [inline internal/service/connect/wait.go:21]: I see these weren't implemented either in that resource. 

I see that Connect Instance does implement the waiter, but doesn't honor the user override from the config:
https://github.com/hashicorp/terraform-provider-aws/blob/f445f5e1980a8b4a6581b2964b8ba70b80330128/internal/service/connect/instance.go#L139

**Review by AdamTylerLynch — CHANGES_REQUESTED** [review]
Requested some changes.

**Comment by GlennChia** [comment]
Re-ran the test cases after most recent commits

```
$ make testacc TESTS=TestAccConnectQuickConnect_serial PKG=connect
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./internal/service/connect/... -v -count 1 -parallel 20 -run='TestAccConnectQuickConnect_serial' -timeout 180m
=== RUN   TestAccConnectQuickConnect_serial
=== RUN   TestAccConnectQuickConnect_serial/disappears
=== RUN   TestAccConnectQuickConnect_serial/basic
--- PASS: TestAccConnectQuickConnect_serial (436.50s)
    --- PASS: TestAccConnectQuickConnect_serial/disappears (141.82s)
    --- PASS: TestAccConnectQuickConnect_serial/basic (294.68s)
PASS
ok      github.com/hashicorp/terraform-provider-aws/internal/service/connect    443.854s
```

File: internal/service/connect/quick_connect.go [hunk 3]
```diff
@@ -0,0 +1,420 @@
...
+			Delete: schema.DefaultTimeout(connectQuickConnectDeletedTimeout),
+		},
+		Schema: map[string]*schema.Schema{
+			"description": {
+				Type:         schema.TypeString,
+				Optional:     true,
+				ValidateFunc: validation.StringLenBetween(1, 250),
+			},
+			"quick_connect_arn": {
```
[the remark this answers is not in the document]
Comment (GlennChia) [reply internal/service/connect/quick_connect.go]: Resolved in ea2f028eecb2d1fd653e18752af768820abe111a

File: internal/service/connect/quick_connect.go [hunk 4]
```diff
@@ -0,0 +1,420 @@
...
+
+	instanceID, quickConnectID, err := QuickConnectParseID(d.Id())
+
+	if err != nil {
+		return diag.FromErr(err)
+	}
+
+	// QuickConnect has 2 update APIs
+	// UpdateQuickConnectConfigWithContext: Updates the configuration settings for the specified quick connect.
+	// UpdateQuickConnectNameWithContext: Updates the name and description of a quick connect.
```
[the remark this answers is not in the document]
Comment (GlennChia) [reply internal/service/connect/quick_connect.go:237]: Resolved in 949778981e1624396ffe670dd2ddeca8a89a2214

File: internal/service/connect/wait.go (hunk 2 above)
Comment (GlennChia) [reply internal/service/connect/wait.go:21]: Resolved in bc0343b81d70556e78c6399bf73e75b1751cf96f by removing timeouts

File: internal/service/connect/quick_connect.go (hunk 1 above)
Comment (GlennChia) [reply internal/service/connect/quick_connect.go]: Resolved in bc0343b81d70556e78c6399bf73e75b1751cf96f by removing timeouts

Comment (GlennChia) [reply internal/service/connect/quick_connect.go]: Missed some unused variables in the earlier commit. Removed them here: c2c12e6a176c8304af7c3261c5c9932868ca6186

**Review by AdamTylerLynch — APPROVED** [review]
LGTM 🚀

**Pull request merged**